homeassistant/components/shelly/coordinator.py

```diff
@@ -673,7 +674,7 @@ async def _async_run_connected_events(self) -> None:
         This will be executed on connect or when the config entry
         is updated.
         """
-        if not self.sleep_period:
+        if not self.sleep_period and self.entry.data.get(CONF_SCRIPT):
```

Comment:
with:
`sleep_period = 0` (valid for xmod)
and
`self.entry.data.get(CONF_SCRIPT) = False` (xmod)

We fall to the `else` and try to setup websocket for a non sleeping device

I think this change is not needed at, it is onyl here to fallback for the `else` to setup websocket for sleeping devices and has no effect for non-sleeping ones even if they do not support scripts.

Reply:
Code updated, tested with new devices and with a WallDisplay installed previously ( no CONF_SCRIPT ).